### PR TITLE
fix(curriculum): prevent passing in case of <img> element not closed

### LINKED ISSUE #54642: The test passes even though the syntax is not correct | Cat Photo App step-8
### Describe the Issue

the challenge is to add `src` attribute to `` element.

you will pass with this solution:
```HTML
<img src = "https://cdn.freecodecamp.org/curriculum/cat-photo-app/relaxing-cat.jpg"
```

which is missing the closing angel bracket `>`.

tried it from a camper topic at the forum here: [Forum](https://forum.freecodecamp.org/t/learn-html-by-building-a-cat-photo-app-step-8/688447)

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-8

### Your code

```HTML
<img src="https://cdn.freecodecamp.org/curriculum/cat-photo-app/relaxing-cat.jpg"
```

**This should not pass**

### Expected behavior

A hint message appears, preventing the camper from passing.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zxc-w**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #54642

### CODE REVIEW

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dc24073f86c76b9248c6ebb.md [hunk 1]
```diff
@@ -45,6 +45,12 @@ Although you have set the `img` element's `src` to the correct URL, it is recomm
 assert(!/\<img\s+src\s*=\s*https:\/\/cdn\.freecodecamp\.org\/curriculum\/cat-photo-app\/relaxing-cat\.jpg/.test(code));
 ```
 
+Your `img` element should be closed either with `>` or `/>`
+
+```js
+assert(/<img\s+src\s*=\s*"https:\/\/cdn\.freecodecamp\.org\/curriculum\/cat-photo-app\/relaxing-cat\.jpg"\s*(?:\/?>)/.test(code));
```
[the remark this answers is not in the document]
Comment (zxc-w) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dc24073f86c76b9248c6ebb.md:51]: I thought of that too when I was trying to check it for the first time, you are correct.

**Review by huyenltnguyen — APPROVED** [review]

**Review by Dario-DC — APPROVED** [review]
LGTM :+1:

**Review by Dario-DC — CHANGES_REQUESTED** [review]
I've just realized that it doesn't take into account single quotes. Should we allow them? @huyenltnguyen

**Comment by zxc-w** [comment]
@huyenltnguyen sure, I was just waiting to see if I should update it or not.

**Review by Dario-DC — APPROVED** [review]

**Pull request merged**